src/odemis/gui/comp/overlay/rectangle.py

```diff
@@ -37,16 +43,60 @@ def __init__(self, rectangle_overlay) -> None:
         self.p_point4 = rectangle_overlay.p_point4
 
 
-class RectangleOverlay(RectanglePointsSelectOverlay, EditableShape):
-    """Overlay representing one rectangle."""
+class RectangleOverlay(EditableShape, RectangleEditingMixin):
```

Comment:
Does RectangleOverlay still matches the "Overlay" interface (ie, it has all the expected attributes and methods). If not, I'd suggest to rename it to not include the word "Overlay". If yes, I'd suggest to have it still inherit from WorldOverlay, just don't call the `__init__()` if that's not helpful.

Reply:
Yes, it still matches the overlay "Overlay" interface. As you said will not include the `__init__()`